src/api/upstreamApi.ts

```diff
@@ -0,0 +1,44 @@
+import { createProductPricingClient, ProductPricingClient } from './upstreams/productPricing';
+import { createContextfulHttpClient, HttpClient } from './httpClient';
+import { ForwardHeaders } from '../context';
+import * as config from '../config'
+
+/**
+ * An umbrella type that groups different upstream services, making it
+ * easy to inject in the Context.
+ *
+ * Then it can be used in functions like
+ * ```typescript
+ * { upstream } // injected into GraphQL resolve function for instance
+ * ...
+ * const quote = await upstream.underwriter.getQuote("id")
+ * ```
+ */
+export interface UpstreamApi {
+    productPricing: ProductPricingClient
+}
+
+export const createUpstreamApi = (
+    token: string,
+    headers: ForwardHeaders
+): UpstreamApi => {
+
+    const createHttpClient = (remotePathPrefix: string, localPort: number): HttpClient => {
+        const allHeaders = (headers as any) as { [key: string]: string }
+
+        let baseUrl: string = `${config.BASE_URL}${remotePathPrefix}`
+        if (config.UPSTREAM_MODE == "local") {
+            baseUrl = `http://localhost:${localPort}`
+            if (config.LOCAL_MEMBERID_OVERRIDE) {
+                allHeaders["Hedvig.token"] = config.LOCAL_MEMBERID_OVERRIDE as string
+            }
+        }
+        return createContextfulHttpClient(baseUrl, token, allHeaders)
+    }
+
+    return {
+        productPricing: createProductPricingClient(
+            createHttpClient("/productPricing", 4085)
```

Comment:
Is hard coding 4085 risky?
```suggestion
            createHttpClient("/productPricing", 4085)
```

Reply:
I personally wouldn't think so? This is only the port used when targeting localhost. Do people usually change the ports of their services locally? In that case, I think we could extend it later to accomodate for that.

If we turn this into an env var or something I think it could give the false indication that it varies, when it is actually quite constant.